app/services/trainees/map_funding_from_dttp.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Trainees
+  class MapFundingFromDttp
+    include ServicePattern
+    include HasDttpMapping
+
+    def initialize(dttp_trainee:)
+      @dttp_trainee = dttp_trainee
+    end
+
+    def call
+      return {} unless dttp_trainee
+
+      funding_attributes.compact
+    end
+
+  private
+
+    attr_reader :dttp_trainee
+
+    def funding_attributes
+      return {} if funding_entity_id.blank?
+
+      if funding_entity_id == Dttp::CodeSets::BursaryDetails::NO_BURSARY_AWARDED
+        return {
+          applying_for_grant: false,
+          applying_for_bursary: false,
+          applying_for_scholarship: false,
+        }
+      end
+
+      if applying_for_new_tier?
+        return {
+          applying_for_grant: false,
+          applying_for_bursary: true,
+          applying_for_scholarship: false,
+          bursary_tier: tier_for_funding,
+        }
+      end
+
+      {
+        applying_for_grant: applying_for_grant?,
+        applying_for_scholarship: applying_for_scholarship?,
+        applying_for_bursary: applying_for_bursary?,
+      }.reject { |_key, value| value == false }
```

Comment:
You know what, I'm not sure.. I didn't change this part 🤔 

cc @archferns  - can you remember why? https://github.com/DFE-Digital/register-trainee-teachers/commit/16f34ab982293121a34d069fbe164351e570ec93

Reply:
It was needed earlier because we were relying on `funding_attributes` being blank to determine if `funding_not_yet_mapped?`, but I believe we can safely remove it as part of this PR. https://github.com/DFE-Digital/register-trainee-teachers/blob/801d2074d550d48a256319dd26d48b566a10dc60/app/services/trainees/create_from_dttp.rb#L497-L499

Reply:
Thanks Arlston that makes a lot of sense! I've removed it 👍